### PR TITLE
Fix notifications for deleted card content blocks

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
When the blocks table started hard deleting blocks, it broke notifications for content block deletions.  This PR fixes that by:

1.  adding new store method to fetch the current version of all card content blocks from history
2. refactor notification code to handle the missing block records in blocks table
3. handle image and file attachment deletion with new text

#### Ticket Link
fixes https://github.com/mattermost/focalboard/issues/4488

### CODE REVIEW

File: server/model/blocktype.go [hunk 1]
```diff
@@ -14,13 +14,16 @@ import (
...
+	TypeUnknown    = "unknown"
+	TypeBoard      = "board"
+	TypeCard       = "card"
+	TypeView       = "view"
+	TypeText       = "text"
+	TypeCheckbox   = "checkbox"
+	TypeComment    = "comment"
+	TypeImage      = "image"
+	TypeAttachment = "attachment"
+	TypeDivider    = "divider"
```
Comment (mgdelacroix) [inline server/model/blocktype.go:26]: <3

File: server/model/blocktype.go [hunk 2]
```diff
@@ -42,6 +45,10 @@ func BlockTypeFromString(s string) (BlockType, error) {
 		return TypeComment, nil
 	case "image":
 		return TypeImage, nil
+	case "attachment":
+		return TypeAttachment, nil
+	case "divider":
+		return TypeDivider, nil
```
Comment (mgdelacroix) [inline server/model/blocktype.go:51]: I think we're missing `checkbox` here

File: server/model/blocktype.go [hunk 3]
```diff
@@ -57,6 +64,8 @@ func BlockType2IDType(blockType BlockType) utils.IDType {
 		return utils.IDTypeView
 	case TypeText, TypeComment:
 		return utils.IDTypeBlock
+	case TypeImage, TypeAttachment:
```
Comment (mgdelacroix) [inline server/model/blocktype.go:67]: Same here

File: server/services/store/sqlstore/blocks.go [hunk 4]
```diff
@@ -42,27 +43,31 @@ func (s *SQLStore) timestampToCharField(name string, as string) string {
 	}
 }
 
-func (s *SQLStore) blockFields() []string {
+func (s *SQLStore) blockFields(alias string) []string {
```
Comment (mgdelacroix) [inline server/services/store/sqlstore/blocks.go:46]: nit: we have a similar but slightly different mechanism on the `boards` store, I'd suggest to pick one and use it on everywhere for consistency.

As for which one to choose, I'm leaning more towards the one proposed on this PR

Comment (wiggin77) [reply server/services/store/sqlstore/blocks.go]: Thanks for reviewing @mgdelacroix.  I modified the boards store to be consistent.

File: server/model/blocktype.go (hunk 3 above)
Comment (wiggin77) [reply server/model/blocktype.go:69]: Fixed

File: server/model/blocktype.go (hunk 2 above)
Comment (wiggin77) [reply server/model/blocktype.go:53]: Fixed

**Review by Rajat-Dabade — APPROVED** [review]
LGTM 👍 , Nice work @wiggin77

**Review by mgdelacroix — APPROVED** [review]
Awesome, thanks @wiggin77 for taking the time to apply the changes!

**Pull request merged**